### PR TITLE
Fixed unit tests runner

### PR DESCRIPTION
**Pull request by wtrocki**
Fix that enables unit tests support. Major refactoring for module will take some time and we still need to have some support for tests on master.I also moved some tests that were not taken into account initially. 
Coverage reports are now generated properly.

@nialldonnellyfh Can You take look?

### CODE REVIEW

**Comment by fheng-build-servers** [comment]
@jasonmadigan @cianclarke please review this PR.

**Comment by fheng-build-servers** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* A component owner should review your change (we've tagged these above)
* This change should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* Ensure JIRA is up to date (e.g. deployment notes, tester notes, fix version, release notes)
* There should be an associated fhcap PR with this change - please provide a link to this

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.

File: test/fixtures/app/fixture_cloud.js [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-})
-.post('/cloud/somefunc')
-.reply(200, cloudReplies.act, cloudReplies.headers)
-.post('/some/custom/cloud/host')
-.reply(200, cloudReplies.act, cloudReplies.headers);
+  .filteringRequestBody(function (){
+    return '*';
+  })
+  .post('/cloud/somefunc')
+  .times(1)
```
Comment (nialldonnellyfh) [inline test/fixtures/app/fixture_cloud.js]: I don't think times is needed if you are only calling it once anyway.

Am I missing something?

File: test/unit/common/test_target.js [hunk 2]
```diff
@@ -1,19 +0,0 @@
-var assert = require('assert');
-var target = require('cmd/fhc/target.js');
-
-module.exports = {
-  'test target': function(cb) {
-    target({_ : []}, function (err) {
-      assert.equal(err, null);
-      return cb();
-      
-      // TODO - not working so well with mocks now that target validates the target REVISIT
```
Comment (nialldonnellyfh) [inline test/unit/common/test_target.js]: Sure you want to delete this file? It does test targeting functionality.

File: test/unit/fh3/cloud/test_cloud.js [hunk 3]
```diff
@@ -16,9 +16,10 @@ module.exports = {
     });
   },
   tearDown : function(cb){
+    require('nock').cleanAll();
```
Comment (nialldonnellyfh) [inline test/unit/fh3/cloud/test_cloud.js]: nock will be a singleton anyway so I don't think you need to do the require in here. Can be at the top of the file like all others.

**Comment by nialldonnellyfh** [comment]
Reviewed. Couple of small comments above. Thanks a lot.

**Comment by wtrocki** [comment]
Applied removed unused statement. Ready to merge.

**Pull request merged**